### ISSUE TITLE
about some details

### ISSUE THREAD

**Issue by dlyldxwl** [VisionLearningGroup/R-C3D#20]
thanks for your job!
now i want to finetune the 135000 caffemodel by using myself dataest,and because of network ,i can't download ActivityNet dataest videos,so i don't know dataest format
i really hope you can help me.thx again!

**Comment by huijuan88** [comment]
Maybe, you can see the other two examples on THUMOS14 and Charades dataset. Their formats are quite similar.

The videos in these two datasets can be downloaded.


On Apr 11, 2018, at 21:39, Chen yh <notifications@github.com<mailto:notifications@github.com>> wrote:


thanks for your job!
now i want to finetune the 135000 caffemodel by using myself dataest,and because of network ,i can't download ActivityNet dataest videos,so i don't know dataest format
i really hope you can help me.thx again!

**Comment by dlyldxwl** [comment]
thank you for you reply.
 I fiuntune 135000 caffemodel, now the network can run. but  I don't understand the information of Terminal output.

Accuracy: 0.984375
I0413 09:36:22.818658  4436 accuracy_layer.cpp:101] Class 0 accuracy : 1
I0413 09:36:22.818661  4436 accuracy_layer.cpp:101] Class 1 accuracy : 0.666667
TRAIN
I0413 09:36:22.826498  4436 solver.cpp:228] Iteration 63, loss = 0.973929
I0413 09:36:22.826509  4436 solver.cpp:244]     Train net output #0: accuarcy = 0.714286
I0413 09:36:22.826514  4436 solver.cpp:244]     Train net output #1: loss_cls = 0.531705 (* 1 = 0.531705 loss)
I0413 09:36:22.826519  4436 solver.cpp:244]     Train net output #2: loss_twin = 0.38329 (* 1 = 0.38329 loss)
I0413 09:36:22.826520  4436 solver.cpp:244]     Train net output #3: rpn_accuarcy = 0.984375
I0413 09:36:22.826522  4436 solver.cpp:244]     Train net output #4: rpn_accuarcy_class = 1
I0413 09:36:22.826524  4436 solver.cpp:244]     Train net output #5: rpn_accuarcy_class = 0.666667
I0413 09:36:22.826527  4436 solver.cpp:244]     Train net output #6: rpn_cls_loss = 0.0271421 (* 1 = 0.0271421 loss)
I0413 09:36:22.826530  4436 solver.cpp:244]     Train net output #7: rpn_loss_twin = 0.0317921 (* 1 = 0.0317921 loss)

1.  I believe the Accuray is rpn_accuarcy, and does the class0 ,1 accuracy indicate that RPN has the accuracy foreground and background classification ?and  Train net output #0: accuarcy = 0.714286 indicate the accuarcy of the R-C3D network?
2.the learing rate of conv1a,conv2a.. are 0, Do you think these layers needs backward computation when finetuned？
3.I don't fine the demo.py file,which means I should write the py file when I want to use a video to detect the effect of caffemodel?

I hope you can answer my questions.
Wish you a happy day, thanks again!

**Comment by dlyldxwl** [comment]
I want to use caffemodel to detect a video, but I can't find the demo file ,can you tell me what i should do?
thanks

**Comment by huijuan88** [comment]
"Class 0 accuracy” indicates the background classification accuracy in one batch. "Class 1 accuracy” is for foreground classification accuracy in one batch.

Opening the lower layers’ learning rate has minor effect, but the training speed is slow down.

The demo file can be written following the test prediction code.




On Apr 12, 2018, at 22:59, Chen yh <notifications@github.com<mailto:notifications@github.com>> wrote:


thank you for you reply.
I fiuntune 135000 caffemodel, now the network can run. but I don't understand the information of Terminal output.

Accuracy: 0.984375
I0413 09:36:22.818658 4436 accuracy_layer.cpp:101] Class 0 accuracy : 1
I0413 09:36:22.818661 4436 accuracy_layer.cpp:101] Class 1 accuracy : 0.666667
TRAIN
I0413 09:36:22.826498 4436 solver.cpp:228] Iteration 63, loss = 0.973929
I0413 09:36:22.826509 4436 solver.cpp:244] Train net output #0: accuarcy = 0.714286
I0413 09:36:22.826514 4436 solver.cpp:244] Train net output #1<https://github.com/VisionLearningGroup/R-C3D/issues/1>: loss_cls = 0.531705 (* 1 = 0.531705 loss)
I0413 09:36:22.826519 4436 solver.cpp:244] Train net output #2<https://github.com/VisionLearningGroup/R-C3D/issues/2>: loss_twin = 0.38329 (* 1 = 0.38329 loss)
I0413 09:36:22.826520 4436 solver.cpp:244] Train net output #3<https://github.com/VisionLearningGroup/R-C3D/issues/3>: rpn_accuarcy = 0.984375
I0413 09:36:22.826522 4436 solver.cpp:244] Train net output #4<https://github.com/VisionLearningGroup/R-C3D/issues/4>: rpn_accuarcy_class = 1
I0413 09:36:22.826524 4436 solver.cpp:244] Train net output #5<https://github.com/VisionLearningGroup/R-C3D/issues/5>: rpn_accuarcy_class = 0.666667
I0413 09:36:22.826527 4436 solver.cpp:244] Train net output #6<https://github.com/VisionLearningGroup/R-C3D/issues/6>: rpn_cls_loss = 0.0271421 (* 1 = 0.0271421 loss)
I0413 09:36:22.826530 4436 solver.cpp:244] Train net output #7<https://github.com/VisionLearningGroup/R-C3D/pull/7>: rpn_loss_twin = 0.0317921 (* 1 = 0.0317921 loss)

  1.  I believe the Accuray is rpn_accuarcy, and does the class0 ,1 accuracy indicate that RPN has the accuracy foreground and background classification ?and Train net output #0: accuarcy = 0.714286 indicate the accuarcy of the R-C3D network?
2.the learing rate of conv1a,conv2a.. are 0, Do you think these layers needs backward computation when finetuned？
3.I don't fine the demo.py file,which means I should write the py file when I want to use a video to detect the effect of caffemodel?

I hope you can answer my questions.
Wish you a happy day, thanks again!

**Comment by dlyldxwl** [comment]
thank you for your reply～
and Train net output #0: accuarcy = 0.714286  indicate R-C3D action classification accuarcy is 0.71?
finally, In test_net.py, i find the input format of network is .pkl, but i want to use a video directly, is there any good method？

**Comment by dlyldxwl** [comment]
@huijuan88 The demo file has been written
and  my dataest is small and video  length is about 10 seconds, can you give me some advice in parameter settings?Or use the network original parameters directly?

**Comment by huijuan88** [comment]
If you just test, you need to follow the origin train parameters.

Since the 10 seconds video is short, maybe a larger fps can be tried to convert the video to more frames.



On Apr 15, 2018, at 03:41, Chen yh <notifications@github.com<mailto:notifications@github.com>> wrote:


@huijuan88<https://github.com/huijuan88> The demo file has been written
and my dataest is small and video length is about 10 seconds, can you give me some advice in parameter settings?Or use the network original parameters directly?

**Comment by dlyldxwl** [comment]
@huijuan88 
my video fps is 120,so I set FPS = 100 and retrain the work,
Train net output #0: accuarcy = 0.963636 , which means the network classification accuracy is 0.96? but i test some videos, result isn't good... can you give me some advices?
and i found a error in activitynet_log_analysis.py, line 113, after call get_segments function, we should set predict_data=[],Otherwise only the detection result of the first video can be output to result.json～

**Comment by huijuan88** [comment]
100fps is too high. Usually, 25fps is maximum. Also, if your dataset is very different from activityNet, it might not work well without training.

I update activitynet_log_analysis.py.



On Apr 16, 2018, at 09:45, Chen yh <notifications@github.com<mailto:notifications@github.com>> wrote:


@huijuan88<https://github.com/huijuan88>
my video fps is 120,so I set FPS = 100 and retrain the work,
Train net output #0: accuarcy = 0.963636 , which means the network classification accuracy is 0.96? but i test some videos, result isn't good... can you give me some advices?
and i found a error in activitynet_log_analysis.py, line 113, after call get_segments function, we should set predict_data=[],Otherwise only the detection result of the first video can be output to result.json～

**Comment by dlyldxwl** [comment]
@huijuan88
Yes. now i am training the network by fiuntuning 13500caffemodel.
My dataest has 192 videos and 3class,  and video length is about 10s, but videos fps are 120, now i want to  finetune 135000caffemodel, can you give me some advice on parameter settings? for example: FPS, batchsize,and so on.
thx!

**Comment by dlyldxwl** [comment]
@sijun-zhou 
hi , I am also a new for action detection.
Because of a competition, i trained this model, after that, i didn't work on it.
I could tell you my computer configuration, 12G memory and TITAN X(pascal).
if you memory isn't enough, i reckon you could use a small batch size or resolution, certainly, it is also work to fix more layers, but which can lead to a bad result.
Above are my suggestion, I can't ensure it can work well.
Finally, I wish you success in your experiment~

**Comment by sijun-zhou** [comment]
@dlyldxwl 
Thanks very much! Really appreciate your reply. I'll look into detail of my problem! :)
BTW, you only use 1 TITAN X card?

Thanks,
Sijun

**Comment by dlyldxwl** [comment]
@sijun-zhou
yes , i just have one TITANX card

**Comment by sijun-zhou** [comment]
@dlyldxwl 
Thank you very much. I'll look into detail of my problem!

**Comment by YanYan0716** [comment]
@dlyldxwl i am a new for deep learning, so could you tell me how to use the caffemodel to detect a video just as you said, thanks a lot, best wish for you

**Comment by viswalal** [comment]
@dlyldxwl , could you please share the demo file you have written?
